app/ui/src/main/java/com/fsck/k9/activity/setup/AccountDiscovery.kt

```diff
@@ -0,0 +1,134 @@
+package com.fsck.k9.activity.setup
+
+import com.fsck.k9.mail.AuthType
+import com.fsck.k9.mail.ConnectionSecurity
+import com.fsck.k9.mail.ServerSettings
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.launch
+import org.jetbrains.anko.coroutines.experimental.bg
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import org.xmlpull.v1.XmlPullParserFactory
+import timber.log.Timber
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+class AccountDiscovery {
```

Comment:
This should probably be at least three different classes. One concerned with downloading, one for parsing the result, and one that takes the parsed result and creates K-9 Mail's ServerSettings objects. Then probably an additional one that ties it all together (the implementation of the interface mentioned below).

Reply:
Okay I've split the Thunderbird code into three classes but I skipped the intermediate step of parsing result and then mapping that to ServerSettings. Do you think it's really necessary? As I'm grabbing only things that we need the intermediate parse result would be 1:1 identical to ServerSettings and it seems it's unlikely it would be necessary to access the parse result in any other context. But I could've overlooked some edge case scenario...